### ISSUE TITLE
Tool to allow adding emojis to the capture

### ISSUE THREAD

**Issue by ChromeQ** [lupoDharkael/flameshot#764]
It would be cool to have a new tool which expands a selection of emojis, and upon clicking it will allow you to click/drag on your capture to put the emoji on the overlay.
Is this something that can be done?

**Comment by pguedes** [comment]
or at least the facepalm one... i really need that! :facepalm:

**Comment by nettum** [comment]
I'm also missing this. I've been using the text and pixelate tools a lot and from time to time I've wanted to add emojis to the capture as well. 

A workaround is to copy emojis from e.g. https://emojipedia.org/ and paste it into the text tool, but it seems like the font does not show all emojis correctly. This workaround also limit you to only one emoji - the one you copied before doing the capture.

**Comment by borgmanJeremy** [comment]
Any thoughts how this would work in practice? When you click the emoji button a box of predefined emojis appear? How many? Maybe someone can sketch a mockup before I code it.

@mmahmoudian  looking for your ideas too.

**Comment by mmahmoudian** [comment]
@borgmanJeremy Actually the way I can see this implemented is relatively simple and using the UI features we have:

A tool that when clicked, opens the side panel listing emojis (and potentially Font Awesome glyphs if user wants). When user selects an item, it highlights the selection to indicate what is going to be used, and then just similar to the incremental counting numbers, user can place them and even control the size. One crucial thing (that Signal's desktop app is also lacking) is search box for emojis/glyphs.

**Comment by borgmanJeremy** [comment]
Yeah integration into the side panel will be most straightforward.

**Comment by nettum** [comment]
Yes, search would be great! I hate picking emojis on facebook (desktop) as it does not have a searchbox.
I like the logic on how the Emoji Selector gnome extension handle emojis. See screenshot at https://github.com/maoschanz/emoji-selector-for-gnome
It has, search, recently used bar (nice to have) and then all the emojis based on categories.

Integration into the side panel sounds good to me.

**Comment by borgmanJeremy** [comment]
I've started working on this but haven't had time to finish it. I need to add skin tone selection and then integrate the widget in Flameshot proper.
![emoji](https://user-images.githubusercontent.com/46930769/132551570-845f7110-5f74-4713-8139-f20501f33f05.gif)

**Comment by fjfnaranjo** [comment]
> I've started working on this but haven't had time to finish it. I need to add skin tone selection and then integrate the widget in Flameshot proper.

Do you need help with this integration? If you send me the code for the selector I can try to do it. If the project owner wants to approve something like this (I saw the "Needs Decision" tag).

**Comment by borgmanJeremy** [comment]
Hey sorry I've been really busy with other flameshot issues and this ended up on the back burner. I made my code public here: https://github.com/borgmanJeremy/emoji

You are totally welcome to help me get it over the line. If you want to work on it let me know here or on our matrix channel: https://matrix.to/#/#Flameshot:feneas.org and we can talk through the remaining items.

Keep in mind this is a rapid prototype still. I killed the first two implementations because they clearly were not going to scale. They were unicode based. 

Off hand the open issues are:

- [ ] emojis don't always vertically scale properly
- [ ] Path to emoji's are hard coded to a fixed path on my hard drive
- [ ] Want to add skin tone selector like signal has
- [ ] Add category menu bar to bottom of window like signal has (categories are already defined and tagged)
- [ ] The code is very hacked, need to take a close look at it now that this implementation seems like its the right path

Then once thats all done we can get it integrated as a Flameshot widget. I want to keep it separate at first so I can easily re-use it on other projects.

**Comment by fjfnaranjo** [comment]
Oops. I thought you already had the code in a branch. If this needs development I don't have the bandwidth now :(

I will keep track of this issue. Maybe I can help in the future.

**Comment by Dont-ask-for-it** [comment]
Locking for the exact same thing!! What's the progress on it?

**Comment by Dont-ask-for-it** [comment]
Is it on the roadmap?

**Comment by mmahmoudian** [comment]
> Is it on the roadmap?

No it is not in the roadmap. Also Jeremy, the dev who commented above and worked on this feature, departed Flameshot as he explained in #2961. I can suggest putting a bounty on the feature after reading the following link:

https://flameshot.org/donate/

**Comment by rklak** [comment]
+1

**Comment by OptionalM** [comment]
In case anyone wants to pick this back up, I'd like to add a suggestion: Don't limit the picker to emoji. People are accustomed to adding "stickers" to their images, which amounts to any png with transparency. Implementing this feature in a way where a user is free to import folders with pngs would allow a lot of creativity. This way the feature also wouldn't need to ship with an up-to-date collection of all emojis, but it could just be any sticker pack. ShareX, a Windows screenshotter, [ships with BlobEmoji](https://github.com/ShareX/ShareX/tree/develop/ShareX.ScreenCaptureLib/Stickers/BlobEmoji).
